iceoryx2-ffi/ffi/src/node_builder.rs

```diff
@@ -107,46 +136,62 @@ impl iox2_node_builder_storage_t {
 ///
 /// # Arguments
 ///
-/// * `node_builder_storage` - Must be either a NULL pointer or a pointer to a valid [`iox2_node_builder_storage_t`]. If it is a NULL pointer, the storage will be allocated on the heap.
+/// * `node_builder_struct_ptr` - Must be either a NULL pointer or a pointer to a valid [`iox2_node_builder_t`]. If it is a NULL pointer, the storage will be allocated on the heap.
 ///
 /// # Returns
 ///
-/// A [`iox2_node_builder_mut_h`] handle to build the actual node.
+/// A [`iox2_node_builder_h`] handle to build the actual node.
 ///
 /// # Safety
 ///
-/// The same [`iox2_node_builder_storage_t`] cannot be used in subsequent calls to this function, unless [`iox2_node_builder_create`] was called before!
+/// The same [`iox2_node_builder_t`] cannot be used in subsequent calls to this function, unless [`iox2_node_builder_create`] was called before!
 #[no_mangle]
 pub unsafe extern "C" fn iox2_node_builder_new(
-    node_builder_storage: *mut iox2_node_builder_storage_t,
-) -> iox2_node_builder_mut_h {
-    let mut handle = node_builder_storage;
-    fn no_op(_storage: *mut iox2_node_builder_storage_t) {}
-    let mut deleter: fn(*mut iox2_node_builder_storage_t) = no_op;
-    if handle.is_null() {
-        handle = iox2_node_builder_storage_t::alloc();
-        deleter = iox2_node_builder_storage_t::dealloc;
+    node_builder_struct_ptr: *mut iox2_node_builder_t,
+) -> iox2_node_builder_h {
+    let mut node_builder_struct_ptr = node_builder_struct_ptr;
+    fn no_op(_: *mut iox2_node_builder_t) {}
+    let mut deleter: fn(*mut iox2_node_builder_t) = no_op;
+    if node_builder_struct_ptr.is_null() {
+        node_builder_struct_ptr = iox2_node_builder_t::alloc();
+        deleter = iox2_node_builder_t::dealloc;
     }
-    debug_assert!(!handle.is_null());
+    debug_assert!(!node_builder_struct_ptr.is_null());
 
-    unsafe {
-        (*handle).deleter = deleter;
-    }
+    (*node_builder_struct_ptr).deleter = deleter;
+    (*node_builder_struct_ptr)
+        .node_builder
+        .init(NodeBuilder::new());
 
-    unsafe {
-        (*handle)
-            .node_builder_maybe_uninit()
-            .write(NodeBuilder::new());
-    }
+    node_builder_struct_ptr as *mut _ as *mut _
+}
+
+/// This function casts an owning [`iox2_node_builder_h`] into a non-owning [`iox2_node_builder_ref_h`]
+///
+/// # Arguments
+///
+/// * `node_builder_handle` obtained by [`iox2_node_builder_new`]
+///
+/// Returns a [`iox2_node_builder_ref_h`]
+///
+/// # Safety
+///
+/// The `node_builder_handle` must be a valid handle.
```

Comment:
What do you think that in the `Safety` section every requirement is a bullet point. This makes it a bit clearer how many requirements I have to ensure, especially when there is more (explanatory) text.